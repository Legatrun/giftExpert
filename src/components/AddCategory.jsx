import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        // para asegurarnos que el input no esta vacio hacemos ete if
        if( inputValue.trim().length <= 1 ) return
        // setCategories((cat) => [inputValue,...cat])
        setInputValue('');
        onNewCategory( inputValue.trim() )
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
} 
