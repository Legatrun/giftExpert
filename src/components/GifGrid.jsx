import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            {
                isLoading && ( <div className="lds-hourglass"></div> )
            }
            <h3>{category}</h3>


            <div className="card-grid" >
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            // se usa spreed operator para exparcir las props de "image"
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired,
}