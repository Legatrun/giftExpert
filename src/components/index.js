// esto es lo primero que lee cuando ponemos "./components"

export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';