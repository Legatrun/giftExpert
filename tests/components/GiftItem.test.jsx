import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Prueba de GiftItem', () => { 
    
    const title = "Saitama"
    const url = "https://api.giphy.com/v1/gifs/search?api_key=OsplqUAzsJ2Mmq8nHpZxg2YJvN8jKMnL&q=rick&limit=10"

    test('should hacer match con el snapshot', () => { 

        const { container } = render( <GifItem title={title} url={url}/> )

        expect( container ).toMatchSnapshot();
     })

     test('should de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render( <GifItem title={title} url={url}/> )
        // screen.debug()
        // expect( screen.getByRole('img').src ).toBe( url )
        // expect( screen.getByRole('img').alt ).toBe( title )

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
      })
      
      test('El titulo debe estar en el title', () => { 
        
        render( <GifItem title={title} url={url}/> )
        expect( screen.getByText( title ) ).toBeTruthy()

       })
 })