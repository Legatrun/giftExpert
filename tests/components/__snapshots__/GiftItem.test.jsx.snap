// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GiftItem should hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://api.giphy.com/v1/gifs/search?api_key=OsplqUAzsJ2Mmq8nHpZxg2YJvN8jKMnL&q=rick&limit=10"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
